continuous-pipe.yml.erb

```diff
@@ -0,0 +1,192 @@
+
+  storage:
+    deploy:
+      services:
+        mysql:
+          specification:
+            volumes:
+              - type: persistent
+                name: database-volume
+                capacity: 5Gi
+                storage_class: default
+
+            volume_mounts:
+              - name: database-volume
+                mount_path: /var/lib/mysql
+
+            command:
+              - /usr/local/bin/docker-entrypoint.sh
+              - mysqld
+              - --ignore-db-dir=lost+found
+              - --max_allowed_packet=128M
+
+            ports:
+              - 3306
+
+            resources:
+              requests:
+                cpu: 50m
+                memory: 250Mi
+              limits:
+                cpu: 500m
+                memory: 2Gi
+
+            environment_variables:
+              MYSQL_PASSWORD: ${DATABASE_PASSWORD}
+              MYSQL_ROOT_PASSWORD: ${DATABASE_ROOT_PASSWORD}
+              MYSQL_DATABASE: ${DATABASE_NAME}
+              MYSQL_USER: ${DATABASE_USER}
+
+  reserve_ip:
+    deploy:
+      services:
+        web:
+          specification:
+            source:
+              image: continuouspipe/landing-page
+              tag: latest
+
+            accessibility:
+              from_external: true
+
+            ports:
+              - 80
+              - 443
+
+      filter:
+        expression: 'tasks.storage.services.database.created'
+
+  setup:
+    run:
+      image:
+        from_service: web
+
+      commands:
+        - container setup
+
+      environment_variables: &WEB_ENV_VARS
+        APP_USER_LOCAL: false
+        DATABASE_NAME: ${DATABASE_NAME}
+        DATABASE_USER: ${DATABASE_USER}
+        DATABASE_PASSWORD: ${DATABASE_PASSWORD}
+        DATABASE_ROOT_PASSWORD: ${DATABASE_ROOT_PASSWORD}
+        AUTH_HTTP_ENABLED: true
+        AUTH_HTTP_HTPASSWD: ${AUTH_HTTP_HTPASSWD}
+        AUTH_IP_WHITELIST_ENABLED: true
+        AUTH_HTTP_WHITELIST_IPS: ${AUTH_HTTP_WHITELIST_IPS}
+
+    filter:
+      expression: 'tasks.storage.services.database.created and code_reference.branch not in ["production", "uat"]'
+
+  deployment:
+    deploy:
+      services:
+        web:
+          specification:
+            accessibility:
+              from_external: true
+
+            environment_variables:
+              <<: *WEB_ENV_VARS
+
+            resources:
+              requests:
+                cpu: 50m
+                memory: 50Mi
+
+              limits:
+                cpu: 1
+                memory: 2G
+
+            ports:
+              - 80
+              - 443
+
+          deployment_strategy:
+            readiness_probe:
+              type: tcp
+              port: 80
+
+            liveness_probe:
+              type: exec
+              command:
+                - curl
+                - -k
+                - -I
+                - -X GET
+                - -H
+                - "Authorization: Basic ${AUTH_HTTP_AUTHORISATION_STRING}"
```

Comment:
It would be good to get 127.0.0.1 whitelisted instead, as it avoids some extra steps to base64 the username/password of basic auth and put it in the UI.